overlord/devicestate/handlers_gadget.go

```diff
@@ -263,8 +259,7 @@ func kernelCommandLineAppendArgs(tsk *state.Task, tr *config.Transaction,
 	return value, nil
 }
 
-func buildAppendedKernelCommandLine(t *state.Task, gd *gadget.GadgetData) (string, error) {
-	st := t.State()
+func buildAppendedKernelCommandLine(t *state.Task, st *state.State, gd *gadget.GadgetData, deviceCtx snapstate.DeviceContext) (string, error) {
```

Comment:
my comment was about `DeviceCtx(st, _t_, nil)`, not passing in st here or use t.State(), I'm fine either way though, but if we pass in st it should be the first argument

Reply:
Ok, I've reverted passing state but kept deviceCtx.